File: frontend/Task Planning App/src/app/types/route.ts
export interface SavedRouteSummary {
  id: string;
  algorithm: string;
  computedAt: string;
}

export interface RouteStopTiming {
  taskId: string;
  position: number;
  travelFromPrevSec?: number;
  arriveTime?: string;       // ISO 8601
  serviceStartTime?: string; // ISO 8601
  serviceEndTime?: string;   // ISO 8601
  waitSec?: number;
}

export interface SavedRouteDetails extends SavedRouteSummary {
  orderedTaskIds: string[];
  stops?: RouteStopTiming[];
  totalDistanceKm?: number;
  totalTravelTimeMin?: number;
  totalDurationMin?: number;
}

File: backend/internal/domain/route/route.go
package route

import "time"

type Route struct {
	ID              string
	UserID          string
	Algorithm       string
	TotalDistanceM  int
	TotalTravelSec  int
	TotalServiceSec int
	TotalWaitSec    int
	ComputedAt      time.Time
}

type Stop struct {
	ID                string
	RouteID           string
	TaskID            string
	Position          int
	TravelFromPrevSec *int
	ArriveTime        *time.Time
	ServiceStartTime  *time.Time
	ServiceEndTime    *time.Time
	WaitSec           *int
}
